### PR TITLE
Content management

### PR DESCRIPTION
**Pull request by TheJarles**
This pull request implements the content view refactor, the routing and the cascading deletion tasks in the project backlog. I also found there was an issue with my earlier implementation of the double reference in UserContent which I fixed. I've done some limited testing locally and found that the cascading deletion works and I've ensured that foreign key constraints indeed render the deletion of contents impossible if they are in a user's bookshelf. The ensure_safe method I've added to the content model might in fact be redundant for this reason, but I left it in just in case.

### CODE REVIEW

File: app/controllers/contents_controller.rb [hunk 1]
```diff
@@ -3,14 +3,22 @@ class ContentsController < ApplicationController
...
   # GET /contents
   def index
-    @contents = Content.all
-
-    render json: @contents
+    render json: {
+      contents: Content.all,
+      content_metadata: ContentMetadatum.all,
+      content_audios: ContentAudio.all,
+      content_resources: ContentResource.all
```
Comment (jabbors) [inline app/controllers/contents_controller.rb]: Does this mean metadata, audio and resources are included in the response in a GET request to `/contents`? If so, the response will be very large and hard to parse, thus I think we shall keep the default behaviour and not include any metadata,audio or resources. That information can be request separately with a GET request to `/contents/<id>`

File: config/routes.rb [hunk 2]
```diff
@@ -5,9 +5,9 @@
   resources :question_inputs
   resources :questions
   resources :question_types
-  resources :content_resources
-  resources :content_metadata
-  resources :content_audios
+  resources :content_resources, except: %i[update show index]
+  resources :content_metadata, except: %i[update show index]
+  resources :content_audios, except: %i[update show index]
```
Comment (jabbors) [inline config/routes.rb]: Let's keep it in mind that we might want to prevent deletes as well. Not really sure yet. I user might accidentally add a resource with incorrect content_id and thus it should be possible to delete it. At the same time I'd like to prevent accidental deletes of resources.

File: app/controllers/contents_controller.rb (hunk 1 above)
Comment (TheJarles) [reply app/controllers/contents_controller.rb]: Seeing as the same method is called for contents, metadata, audios and resources, I would assume this is the case. I'll revert the code back to the default behavior as requested.

File: config/routes.rb (hunk 2 above)
Comment (TheJarles) [reply config/routes.rb]: I'll leave deleting in for now, but perhaps I could add a comment that delete should be prohibited at some later point?

File: app/controllers/contents_controller.rb (hunk 1 above)
Comment (jabbors) [reply app/controllers/contents_controller.rb]: 👍

**Comment by jabbors** [comment]
Will you close this when #13 is merged?

**Comment by TheJarles** [comment]
> Will you close this when #13 is merged?
Well I could, but this pull request has changes that are not present in #13. If you want I could merge this branch with #13 and close this pull once #13 is merged.

**Comment by jabbors** [comment]
I only saw one commit that was not part of #13 which is the review additions you added. I suggest that you cherry-pick that commit into #13, close this and then we forget about it 😄
